### DISCUSSION TITLE
Is there a way to verify a single message with only an aggregate signature and aggregate public key?

### DISCUSSION THREAD

**Discussion by andyraddatz** [paulmillr/noble-bls12-381#40] [General]
Hello and thank you for this library!  It's really helping me learn about ECC and BLS.  I have a use case in mind that does not seem to work, and I am not sure if I am doing it wrong, or if it is just not possible.

A simple addition to your example [here](https://github.com/paulmillr/noble-bls12-381#usage) illustrates what I'm trying to do:

````javascript
  // aggSignature3 = 3 different messages signed with 3 different private keys (3 diff individual public sigs) aggregated into one sig
  // aggPubKey2 = 3 public keys corresponding to the 3 private keys above aggregated into one pub key

  // any way to verify each individual message from the group of 3 with only the aggregated sigs and pub keys? or impossible?
  const onlyFirstMessage = messages[0];
  const isCorrect4 = await bls.verify(aggSignature3, onlyFirstMessage, aggPubKey2) // str, str, str
  console.log('is correct: ', isCorrect4); // returns false
````

**Comment by paulmillr** [comment]
Try `verifyBatch`?

**Reply by andyraddatz** [reply]
returns `UnhandledPromiseRejectionWarning: Error: Pubkey count should equal msg count` if I just swap `verify` with `verifyBatch`
